File: packages/functions/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/functions/classes/options/CallByUrlOptions.java
package io.capawesome.capacitorjs.plugins.firebase.functions.classes.options;

import androidx.annotation.NonNull;

import java.net.MalformedURLException;
import java.net.URL;

public class CallByUrlOptions extends CallOptions {

    @NonNull
    private URL url;

    public CallByUrlOptions(@NonNull String url, Object data) throws MalformedURLException {
        super(data);
        this.url = new URL(url);
    }

    @NonNull
    public URL getUrl() {
        return url;
    }
}

File: packages/functions/ios/Plugin/Classes/Options/CallByUrlOptions.swift
import Foundation

@objc public class CallByUrlOptions: CallOptions {
    private var url: URL

    init(url: String, data: Any?, timeout: Int?) {
        self.url = URL(string: url)!
        super.init(data: data, timeout: timeout)
    }

    func getUrl() -> URL {
        return url
    }
}
